payments-core/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -110,38 +110,40 @@ internal class DefaultFlowController internal constructor(
     private val viewModel =
         ViewModelProvider(viewModelStoreOwner)[FlowControllerViewModel::class.java]
 
-    // The properties below are lazily initialized to allow the developer to set the publishableKey
-    // and stripeAccountId in PaymentConfiguration any time before configuring the FlowController
-    // through configureWithPaymentIntent or configureWithSetupIntent.
-
-    private val paymentConfiguration: PaymentConfiguration by lazy {
+    // The provider below always refreshes the PaymentConfiguration instance to allow the developer
+    // to set the publishableKey and stripeAccountId in PaymentConfiguration any time before
+    // configuring the FlowController through configureWithPaymentIntent or
+    // configureWithSetupIntent.
+    // TODO(ccen) inject paymentConfigurationProvider
+    private val paymentConfigurationProvider = Provider {
+        PaymentConfiguration.clearInstance()
```

Comment:
Is this needed? If `init` is called the instance is updated and `getInstance` will return the new one, with the new values.

Reply:
Oh you're right, `init` will overwrite the instance, removed the `clearInstance` call